test/test_nn.py

```diff
@@ -8391,6 +8391,41 @@ def test_rnn_retain_variables(self, device, dtype):
             with torch.backends.cudnn.flags(enabled=False):
                 self._test_rnn_retain_variables(device, dtype)
 
+    @onlyCUDA
```

Comment:
How much memory do those tests require? Regular CI machines are M60s with 6GB memory IIRC, so if you need more, use large memory decorator.

Reply:
Looking at the size of input + output, it takes 0.625GB. Given the way input/output are allocated in the code path, it should have cached 0.75GB with 0.625GB actually allocated.

Running the test independently I'm seeing 2.5GB cached memory in total at the end of the test, so we should be safe here.